### PR TITLE
fix(content-releases): Limit min date selection to schedule a release

### PR DESCRIPTION
**Pull request by madhurisandbhor**
### What does it do?

Min date considering selected timezone added for scheduling.

### Why is it needed?

https://strapi-inc.atlassian.net/browse/CS-627

### How to test it?
- Try to to create a scheduled release and check that you should not be allowed to select date past today. 
- Also try to check with timezones with `UTC-HH:00`, should display correct minimun date for selected timezone.
- Also BE validation should throw if you select a time earlier in the day which is passed.

### CODE REVIEW

**Review by simotae14 — APPROVED** [review]
It looks good to me, I have also tried all the possible scenario and even the BE validation works well.

File: packages/core/content-releases/admin/src/components/ReleaseModal.tsx [hunk 1]
```diff
@@ -255,12 +256,12 @@ const getTimezones = (selectedDate: Date) => {
...
 
-    // Offset and timezone are concatenated with '_', so to split and save the required timezone in DB
-    return { offset: utcOffset, value: `${utcOffset}_${timezone}` } satisfies ITimezoneOption;
+    // Offset and timezone are concatenated with '&', so to split and save the required timezone in DB
+    return { offset: utcOffset, value: `${utcOffset}&${timezone}` } satisfies ITimezoneOption;
   });
 
   const systemTimezone = timezoneList.find(
-    (timezone) => timezone.value.split('_')[1] === Intl.DateTimeFormat().resolvedOptions().timeZone
+    (timezone) => timezone.value.split('&')[1] === Intl.DateTimeFormat().resolvedOptions().timeZone
```
Comment (joshuaellis) [inline packages/core/content-releases/admin/src/components/ReleaseModal.tsx:264]: I would have made `'&'` a constant to make it clear what that is 😄 @madhurisandbhor

Comment (madhurisandbhor) [reply packages/core/content-releases/admin/src/components/ReleaseModal.tsx:264]: I knew you will have something to say thats why I requested your approval 😄 and I waited. 
I have added a comment above though why concatenating with '&'.

Comment (joshuaellis) [reply packages/core/content-releases/admin/src/components/ReleaseModal.tsx:264]: hahaha sorry, busy day, next time ping me on slack lol i dont mind 😄

**Pull request merged**